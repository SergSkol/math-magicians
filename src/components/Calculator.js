/* eslint class-methods-use-this: ["error", { "exceptMethods": ["render", "CreateButtons"] }] */
/* eslint-env es6 */
import React from 'react';
import arrButtons from './Buttons';
import calculate from '../logic/calculate';

export default class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      operation: null,
      next: null,
    };
    this.handleClick = this.handleClick.bind(this);
    this.CreateButtons = this.CreateButtons.bind(this);
  }

  handleClick(event) {
    const result = calculate(this.state, event.target.innerHTML);
    this.setState(result);
  }

  CreateButtons() {
    const btns = [];
    arrButtons.forEach((btn) => {
      btns.push(<button
        key={btn.key}
        className={btn.class}
        onClick={this.handleClick}
        >{btn.show}</button>);
    });
    return (btns);
  }

  render() {
    return (
      <div className="container">
        <div className="result">
          <p> {this.state.total}</p>
          <p> {this.state.operation}</p>
          <p> {this.state.next}</p>
        </div>
          <this.CreateButtons />
      </div>
    );
  }
}
