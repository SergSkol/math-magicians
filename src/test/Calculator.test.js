import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('Calculator component snapshots test', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
