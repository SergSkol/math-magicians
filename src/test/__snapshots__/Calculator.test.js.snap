// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component snapshots test 1`] = `
<div
  className="calcContent"
>
  <h2>
    Let\`s do some math!
  </h2>
  <div
    className="container"
  >
    <div
      className="result"
    >
      <p>
         
        0
      </p>
      <p>
         
      </p>
      <p>
         
      </p>
    </div>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      AC
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      +/-
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      %
    </button>
    <button
      className="btn-highlight"
      onClick={[Function]}
    >
      รท
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      7
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      8
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      9
    </button>
    <button
      className="btn-highlight"
      onClick={[Function]}
    >
      x
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      4
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      5
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      6
    </button>
    <button
      className="btn-highlight"
      onClick={[Function]}
    >
      -
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      1
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      2
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      3
    </button>
    <button
      className="btn-highlight"
      onClick={[Function]}
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
    >
      0
    </button>
    <button
      className="btn-standart"
      onClick={[Function]}
    >
      .
    </button>
    <button
      className="btn-highlight"
      onClick={[Function]}
    >
      =
    </button>
  </div>
</div>
`;
